### PR TITLE
Check allocation errors in wlr_output_layout

### PR DESCRIPTION
**Pull request by emersion**

### CODE REVIEW

File: types/wlr_output_layout.c [hunk 1]
```diff
@@ -8,14 +8,14 @@
 #include <assert.h>
 
 struct wlr_output_layout_state {
-	struct wlr_box *_box;
```
Comment (acrisci) [inline types/wlr_output_layout.c]: I put an underscore here to emphasize that you should never read this, but only use the getter.

File: types/wlr_output_layout.c [hunk 2]
```diff
@@ -52,25 +57,24 @@ void wlr_output_layout_destroy(struct wlr_output_layout *layout) {
 
 	wl_signal_emit(&layout->events.destroy, layout);
 
-	struct wlr_output_layout_output *_output, *temp = NULL;
-	wl_list_for_each_safe(_output, temp, &layout->outputs, link) {
-		wlr_output_layout_output_destroy(_output);
+	struct wlr_output_layout_output *output, *temp = NULL;
```
Comment (acrisci) [inline types/wlr_output_layout.c]: should be `l_output` for consistency with the rest of the code.

File: types/wlr_output_layout.c (hunk 1 above)
Comment (emersion) [reply types/wlr_output_layout.c]: Oh, okay, I'll add a comment too then.

**Comment by acrisci** [comment]
👍

**Pull request merged**